File: incomplete_fizz_buzz/php/lib/Looper.php
<?php

class Looper
{

  public function __construct($fizz_buzz_runner, $inputer)
  {
    $this->fizz_buzz_runner = $fizz_buzz_runner;
    $this->inputer = $inputer;
  }

  public function loop()
  {
    while(true) {
      $selector = $this->inputer->gets();
      if($selector === '0') {
        return;
      }
      $this->fizz_buzz_runner->run($selector);
    }
  }
}

File: incomplete_fizz_buzz/java/src/Looper.java
public class Looper {

  private FizzBuzzRunner fizzBuzzRunner;
  private Inputable inputer;

  public Looper(FizzBuzzRunner fizzBuzzRunner, Inputable inputer) {
    this.fizzBuzzRunner = fizzBuzzRunner;
    this.inputer = inputer;
  }

  public void loop() {
    System.out.println(FizzBuzzRunner.DISCRIPTION_TEXT);
    while(true) {
      String selector = inputer.gets();
      if(selector.equals("0")) {
        System.out.println("プログラムを終了します．");
        return;
      }
      fizzBuzzRunner.run(selector);
    }
  }
}
